docs/tutorials/jsx-transform-migration.md

```diff
@@ -0,0 +1,477 @@
+---
+id: jsx-transform-migration
+title: Transitioning to the New JSX Transform
+description: A guide to migrating your project to the New JSX Transform
+---
+
+The Backstage core libraries are in the process of deprecating React 16 and evaluating the adoption of React 19. In React 19, the [New JSX Transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html), introduced in September 2020, is now [mandatory](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#installing). This requires a modification in how React is imported into components to ensure compatibility with React 19.
```

Comment:
Just wondering as it might confuse some people (like me 😆) we should be safe to be on React 18, but before we move to 19 we should do this? When we say that we are evaluating the adoption of React 19 does that potentially mean I shouldn't jump the gun and do these steps?

Just trying to wrap my head around should I be doing this now? or should I wait for an all green 🤔